api/app/c_haines/severity_index.py

```diff
@@ -0,0 +1,453 @@
+            c_haines_index=get_severity_string(feature['properties']['severity']),
+            c_haines_prediction=prediction)
+        # Add to current session.
+        session.add(polygon)
+    # Only commit once we have everything.
+    session.commit()
+
+
+def generate_severity_data(c_haines_data):
+    """ Generate severity index data, iterating over c-haines data.
+    NOTE: Iterating to generate c-haines, and then iterating again to generate severity is a bit slower,
+    but results in much cleaner code.
+    """
+    logger.info('Generating c-haines severity index data.')
+    severity_data = []
+    mask_data = []
+    for row in c_haines_data:
+        severity_row = []
+        mask_row = []
+        for cell in row:
+            severity = get_severity(cell)
+            severity_row.append(severity)
+            # We ignore severity 0.
+            if severity == 0:
+                mask_row.append(0)
+            else:
+                mask_row.append(1)
+        severity_data.append(severity_row)
+        mask_data.append(mask_row)
+    return numpy.array(severity_data), numpy.array(mask_data)
+
+
+class EnvCanadaPayload():
+    """ Handy class to store payload information in . """
+
+    def __init__(self):
+        self.filename_tmp_700: str = None
+        self.filename_tmp_850: str = None
+        self.filename_dew_850: str = None
+        self.prediction_timestamp: datetime = None
+        self.model_run: CHainesModelRun = None
+
+
+class CHainesSeverityGenerator():
+    """ Class responsible for orchestrating the generation of Continous Haines severity
+    index polygons.
+
+    Steps for generation of severity level as follows:
+    1) Download grib files.
+    2) Iterate through raster rows, generating an in memory raster containing c-haines severity indices.
+    3) Turn raster data into polygons, storing in intermediary GeoJSON file.
+    4) Write polygons to database.
+    """
+
+    def __init__(self, model: ModelEnum, projection: ProjectionEnum, session: Session):
+        self.model = model
+        self.projection = projection
+        self.c_haines_generator = CHainesGenerator()
+        self.session = session
+
+    def _collect_payload(self,
+                         urls: dict,
+                         prediction_timestamp: datetime,
+                         model_run: CHainesModelRun,
+                         tmp_path: str) -> Union[EnvCanadaPayload, None]:
+        """ Collect all the different things that make up our payload: our downloaded files,
+        model run, and prediction timestamp. """
+
+        def _download_files(urls: dict,
+                            model: ModelEnum,
+                            tmp_path: str) -> Union[List[str], None]:
+            """ Try to download all the files """
+            filenames = []
+            for key in make_model_levels(model):
+                # Try to download this file.
+                filename = download(urls[key], tmp_path)
```

Comment:
it's not immediately clear whether `tmp_path` is short for "temporary path" or something related to temperature...?

Reply:
Changed variable name.